### DISCUSSION TITLE
How to run Qodana per module for multi module projects (Gradle Plugin)

### DISCUSSION THREAD

**Discussion by danielhalim11** [JetBrains/Qodana#78] [Q&A] [answered]
Hi, is it possible to run qodana per module for multi module projects using qodana gradle plugin?
for example: `./gradlew module_a:runInspections`

I have tried to apply the plugin to module_a, but I got this error:
- Qodana download the wrong gradle version with the one I specified
- The error message seems showing wrong gradle path, `Caused by: org.gradle.internal.exceptions.LocationAwareException: Build file '/data/project/build.gradle' line: 2`

Thank you.

**Comment by brichbash** [comment, marked as answer]
Hi!
Sorry for the delay.
It's not possible to run the gradle plugin per module. As you've noticed, the plugin mounts only the chosen module to the docker container and thus it causes the exception. You could use `qodana.yaml`([docs](https://www.jetbrains.com/help/qodana/qodana-yaml.html#Set+up+a+profile)) to exclude all unwanted folders from analysis scope but still run the plugin from the root.

**Reply by danielhalim11** [reply]
ok, noted. Thanks for the explanation.
